courtlistener/process_cl.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import csv
+import logging
+import os
+import re
+import sys
+
+import pandas as pd
+
+from licensed_pile.licenses import PermissiveLicenses
+from licensed_pile.logs import configure_logging
+from licensed_pile.write import to_dolma
+
+SOURCE_NAME = "CourtListenerOpinion"
+
+csv.field_size_limit(sys.maxsize)
+
+logger = configure_logging("court-listener-opinion")
+
+
+def process_court_listener(file_path):
+    df = pd.read_csv(file_path)
+
+    # add metadata column
+    df["metadata"] = str(PermissiveLicenses.PD)
+
+    # add source column
+    df["source"] = SOURCE_NAME
+
+    """
+    Remove columns:
+    date_modified
+    author_str
+    per_curiam
+    joined_by_str
+    type
+    sha1
+    page_count
+    local_path
+    extracted_by_ocr
+    author_id
+    cluster_id
+    """
+    df = df.drop(
+        columns=[
+            "date_modified",
+            "author_str",
+            "per_curiam",
+            "joined_by_str",
+            "type",
+            "sha1",
+            "page_count",
+            "local_path",
+            "extracted_by_ocr",
+            "author_id",
+            "cluster_id",
+        ]
+    )
+
+    """
+    Merge columns based on Court Listener documentation:
+    html_with_citations
+    html_columbia
+    html_lawbox
+    xml_harvard
+    html_anon_2020
+    html
+    plain_text
+    """
+    df["text"] = (
+        df["html_with_citations"]
+        .combine_first(df["html_columbia"])
+        .combine_first(df["html_lawbox"])
+        .combine_first(df["xml_harvard"])
+        .combine_first(df["html_anon_2020"])
+        .combine_first(df["html"])
+    )
+
+    # keep only the text columns and drop null values
+    df = df.drop(
+        columns=[
+            "html",
+            "html_anon_2020",
+            "html_lawbox",
+            "html_columbia",
+            "xml_harvard",
+            "html_with_citations",
+        ]
+    ).dropna(subset=["text"])
```

Comment:
Drop NA is a very different operation than dropping columns. We should separate them into two different lines so the dropna doesn't get lost in a refactor.

Reply:
Will do and add comments.

Reply:
How about something like this:
```python
    # drop the columns used in combine_first to avoid redundancy
    df = df.drop(
        columns=[
            "html",
            "html_anon_2020",
            "html_lawbox",
            "html_columbia",
            "xml_harvard",
            "html_with_citations",
        ]
    )

    # drop null values in the text column
    df = df.dropna(subset=["text"])
```